### ISSUE TITLE
Compilation failure with ghc 7.4.2

### ISSUE THREAD

**Issue by edsko** [lpsmith/bytestring-builder#11]
Not sure if this still matters (old ghc after all), but on ghc 7.4.2 building fails with:

```
src/Data/ByteString/Builder/Prim.hs:553:13:
    Illegal bang-pattern (use -XBangPatterns):
    ! br'
```

Bizarrely, if we compile with `--builddir=foo` compilation succeeds! I've opened a ghc ticket about this at https://ghc.haskell.org/trac/ghc/ticket/12655 , but if you care about old ghc versions you might want to work-around this.

**Comment by lpsmith** [comment]
Strange indeed!

Definitely interested in a fix,  though it'd be nice to know where #10 is at,  and as always, it'd be nice to regenerate bytestring-builder from upstream rather than to fix it here.

**Comment by edsko** [comment]
See the ghc ticket for some references about what's going on here; it's a bug in ghc's lexer. A workaround is declaring `BangPatterns` globally (which is how I managed to build this in the end).

**Comment by RyanGlScott** [comment]
I believe this is fixed as of `bytestring-builder-0.10.8.2.0`. Please re-open if you still experience this issue.
